packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactStylesDiffMap.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+package com.facebook.react.uimanager
+
+import com.facebook.infer.annotation.Nullsafe
+import com.facebook.react.bridge.Dynamic
+import com.facebook.react.bridge.ReadableArray
+import com.facebook.react.bridge.ReadableMap
+
+/**
+ * Wrapper for [ReadableMap] which should be used for styles property map. It extends some of
+ * the accessor methods of [ReadableMap] by adding a default value property such that caller
+ * is enforced to provide a default value for a style property.
+ *
+ *
+ * Instances of this class are used to update [View] or [CSSNode] style properties.
+ * Since properties are generated by React framework based on what has been updated each value in
+ * this map should either be interpreted as a new value set for a style property or as a "reset this
+ * property to default" command in case when value is null (this is a way React communicates change
+ * in which the style key that was previously present in a map has been removed).
+ *
+ *
+ * NOTE: Accessor method with default value will throw an exception when the key is not present
+ * in the map. Style applicator logic should verify whether the key exists in the map using [ ][.hasKey] before fetching the value. The motivation behind this is that in case when the updated
+ * style diff map doesn't contain a certain style key it means that the corresponding view property
+ * shouldn't be updated (whereas in all other cases it should be updated to the new value or the
+ * property should be reset).
+ */
+@Nullsafe(Nullsafe.Mode.LOCAL)
+public class ReactStylesDiffMap(props: ReadableMap) {
+
+  @JvmField
+  public val mBackingMap: ReadableMap = props
+
+  public fun toMap(): Map<String, Any?> = mBackingMap.toHashMap()
```

Comment:
The original signature in Java, did not have this second argument as nullable. Can we try keeping it as it was?

```suggestion
  public fun toMap(): Map<String, Any> = mBackingMap.toHashMap()
```

Reply:
@mateoguzmana Gives 
Return type mismatch: expected 'kotlin.collections.Map<kotlin.String, kotlin.Any>', actual 'java.util.HashMap<kotlin.String, kotlin.Any?>' 
On making public fun toMap(): Map<String, Any> = mBackingMap.toHashMap()

Reply:
We can leave it as-is then